examples/with-netlify-cms/pages/blog/post/[slug].js

```diff
@@ -24,12 +24,29 @@ const Post = ({ blogpost }) => {
   )
 }
 
-Post.getInitialProps = async ({ query }) => {
-  const { slug } = query
-  const blogpost = await import(`../../../content/blogPosts/${slug}.md`).catch(
-    () => null
-  )
-  return { blogpost }
+export async function getStaticPaths() {
+  const markdownFiles = require
+    .context('../../../content/blogPosts', false, /\.md$/)
+    .keys()
+    .map(relativePath => relativePath.substring(2))
+
+  const paths = markdownFiles.map(path => ({
+    params: {
+      slug: path,
+    },
+  }))
+
+  return { paths, fallback: false }
+}
+
+export async function getStaticProps({ params }) {
+  const blog = await import(
+    `../../../content/blogPosts/${params.slug}.md`
+  ).catch(() => null)
+
+  let blogpost = JSON.stringify(blog)
+  blogpost = JSON.parse(blogpost)
```

Comment:
What is the `stringify` and `parse` here for?

Reply:
@lfades It gives error `object [object Module] cannot be serialized`, so i made blogpost a plain object

Also I am not so sure whether this example should use `getStaticProps` or `getServerSideProps`, if I use `getStaticProps`, refreshing blog page will give 404.

Reply:
That's an issue, we can't accept `getServerSideProps` because this is a CMS example, `getStaticProps` should be the way to go